service-worker/foreign-fetch/foreign-fetch-sw.js

```diff
@@ -0,0 +1,21 @@
+// This is deployed to https://foreign-fetch-demo.appspot.com/foreign-fetch-sw.js
+
+self.addEventListener('install', event => {
+  event.registerForeignFetch({
+    scopes: ['/random'], // or self.registration.scope to handle everything
+    origins: ['*'] // or ['https://client1.com'] to limit the remote origins
```

Comment:
I would replace `https://client1.com` with `<client-1 origin-domain>` or something that isn't tied to a domain that might exist (there doesn't appear to be anything at client1.com right now in case anyone was wondering :P)